From be44d0251a2540f3b8d7205e0bc6659704366711 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 30 Jan 2025 00:39:30 +0100
Subject: [PATCH] arm64: dts: qcom: ipq9574: add QPIC SPI NAND default
 partition nodes

Add QPIC SPI NAND default partition nodes for RDP reference board.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 .../boot/dts/qcom/ipq9574-rdp-common.dtsi     | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq9574-rdp-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574-rdp-common.dtsi
@@ -189,6 +189,34 @@
 		nand-ecc-engine = <&qpic_nand>;
 		nand-ecc-strength = <4>;
 		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:training";
+				reg = <0x0 0x80000>;
+				read-only;
+			};
+
+			partition@80000 {
+				label = "0:license";
+				reg = <0x80000 0x40000>;
+				read-only;
+			};
+
+			partition@c0000 {
+				label = "rootfs";
+				reg = <0xc0000 0x3c00000>;
+			};
+
+			partition@3cc0000 {
+				label = "rootfs_1";
+				reg = <0x3cc0000 0x3c00000>;
+			};
+		};
 	};
 };
 
